tests/rptest/tests/datalake/custom_partitioning_test.py

```diff
@@ -215,3 +230,279 @@ def test_basic(self, cloud_storage_type, catalog_type):
                     f"select file_path from {table_name}.files"))
             assert len(files_after) == partitions
             assert files_after.issubset(files_before)
+
+    @cluster(num_nodes=6)
+    @matrix(cloud_storage_type=supported_storage_types(),
+            catalog_type=supported_catalog_types())
+    def test_spec_evolution(self, cloud_storage_type, catalog_type):
+        with DatalakeServices(self.test_context,
+                              redpanda=self.redpanda,
+                              catalog_type=catalog_type,
+                              include_query_engines=[QueryEngineType.SPARK
+                                                     ]) as dl:
+            # Create an iceberg topic with a custom partition spec and produce
+            # some data.
+
+            topic_name = "foo"
+            msg_count = 1000
+            partitions = 5
+            dl.create_iceberg_enabled_topic(
```

Comment:
@ztlpn create_iceberg_enabled_topic() silently overrides target_lag_ms to 60s per topic (hence overriding cluster 
 config iceberg_target_lag_ms), try passing in `target_lag_ms` param to this method instead?

Reply:
Oh this explains why it is working for one topic and not the other! I guess I like it better when it is set explicitly by the test. Overriding it to 10s is okay, as long as the value is small enough for wait helpers to work and not timeout.